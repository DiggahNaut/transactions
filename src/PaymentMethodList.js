import React from 'react';

const PaymentMethodList = (props) => {
    const paymentMethodList = props.paymentData.map((row, index) => {
        return <li className="list-group-item" key={index}> {row.label} <span className="badge badge-primary badge-pill">{row.amountOfUse}</span> </li>;
    });
    
    return (
        <div className="text-center">
            <h3>Рейтинг популярности платёжных систем.</h3>
            <div>
                Рейтинг строится на основе всех совершенных транзакций платежной системой, независимо от статуса.
                Рейтинг идёт сверху вниз: от часто используемого к редко используемому.
            </div>

            <ul className="list-group">
                {paymentMethodList}
            </ul>
        </div>
    );
}

export default PaymentMethodList;