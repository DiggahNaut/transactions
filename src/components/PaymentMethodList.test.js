import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import PaymentMethodList from './PaymentMethodList';

// Возможно стоило использовать getPaymentData функцию и передать transactions.json.
// Поскольку данные статические, решил использовать "заглушку" в виде переменной paymentData.
var paymentData = [
    {"amountOfUse": 71, "label": "Credit/Debit Cards"},
    {"amountOfUse": 15, "label": "PayPal"},
    {"amountOfUse": 8, "label": "Your Balance"},
    {"amountOfUse": 2, "label": "Webmoney"},
    {"amountOfUse": 2, "label": "Google Pay"},
    {"amountOfUse": 1, "label": "QIWI"},
    {"amountOfUse": 1, "label": "RAZER zGOLD"},
    {"amountOfUse": 1, "label": "MobileGo"}
];

// В этом компоненте происходит только рендер списка. Решил что снапшота и простого рендера без краша должно хватить.
it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PaymentMethodList paymentData={paymentData}/>, div);
});

it("check PaymentMethodList snapshot", () => {
    const PaymentMethodListComponent = renderer.create(<PaymentMethodList paymentData={paymentData}/>).toJSON();
    expect(PaymentMethodListComponent).toMatchSnapshot();
});