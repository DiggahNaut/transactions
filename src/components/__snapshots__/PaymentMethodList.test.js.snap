// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check PaymentMethodList snapshot 1`] = `
<div
  className="text-center"
>
  <h3>
    Рейтинг популярности платёжных систем.
  </h3>
  <div>
    Рейтинг строится на основе всех совершенных транзакций платежной системой, независимо от статуса. Рейтинг идёт сверху вниз: от часто используемого к редко используемому. В синем кружке отмечено количество транзакций.
  </div>
  <ul
    className="list-group list"
  >
    <li
      className="list-group-item"
    >
      Credit/Debit Cards
      <span
        className="badge badge-primary badge-pill"
      >
        71
      </span>
    </li>
    <li
      className="list-group-item"
    >
      PayPal
      <span
        className="badge badge-primary badge-pill"
      >
        15
      </span>
    </li>
    <li
      className="list-group-item"
    >
      Your Balance
      <span
        className="badge badge-primary badge-pill"
      >
        8
      </span>
    </li>
    <li
      className="list-group-item"
    >
      Webmoney
      <span
        className="badge badge-primary badge-pill"
      >
        2
      </span>
    </li>
    <li
      className="list-group-item"
    >
      Google Pay
      <span
        className="badge badge-primary badge-pill"
      >
        2
      </span>
    </li>
    <li
      className="list-group-item"
    >
      QIWI
      <span
        className="badge badge-primary badge-pill"
      >
        1
      </span>
    </li>
    <li
      className="list-group-item"
    >
      RAZER zGOLD
      <span
        className="badge badge-primary badge-pill"
      >
        1
      </span>
    </li>
    <li
      className="list-group-item"
    >
      MobileGo
      <span
        className="badge badge-primary badge-pill"
      >
        1
      </span>
    </li>
  </ul>
</div>
`;
